test/test_python_dispatch.py

```diff
@@ -1839,5 +1839,13 @@ def test_basic(self):
         r = torch._C._EnablePythonDispatcher()
         torch.add(x, x)
 
+    def test_lstsq(self):
+        a = torch.randn(4, 3)
+        b = torch.rand(4, 3)
+        expected_shape = torch.linalg.lstsq(a, b).solution.shape
+        r = torch._C._EnablePythonDispatcher()
```

Comment:
Should it be wrapped in `try: finally: del r`? I'm not sure if there's a possibility to leak it into other tests.

Reply:
I think it's an RAII object, see `enable_python_dispatch`: https://github.com/pytorch/pytorch/blob/master/torch/_dispatch/python.py#L16